### DISCUSSION TITLE
Generating events on motion within zones

### DISCUSSION THREAD

**Discussion by klutzzykitty** [blakeblackshear/frigate#4754] [Ask A Question] [answered]
Hello ! 
I was trying to figure out how to enable events to be generated upon motion detection happening within a zone. 

Is there a way to generate an event based on motion being detected, irrespective of what object it is? 
Went through the docs explaining the  [different modes and configuration](https://docs.frigate.video/configuration/record/#what-do-the-different-retain-modes-mean) but couldn't get it. 

`  events:
    # Optional: Number of seconds before the event to include (default: shown below)                               
     retain:
       # Required: Default retention days (default: shown below)                                                      
       mode: motion`

I suppose the above part only effects what gets recorded and not the events generated. 

Below is my zone and surrounding motion masks. 
![cam_view](https://user-images.githubusercontent.com/118447577/208675120-465efae6-1831-438f-b234-72b141d0d277.png)

My goal is to generate an alert upon some motion being detected within the zone. 
As far as I've understood Frigate, the events and snapshots are only generated by detected objects isn't it ? 
And motion is just used as a primary check to send only that portion of the frame for object detection. 

So, what would an example _config.yml_ look like if one wants to just detect motion within a zone? 
And then maybe generate a snapshot  and video of it ?

**Comment by NickM-27** [comment]
Motion is really not reliable as it's just a change in pixels, lights get turned on and the entire stream is one big glob of motion.

Zones are only applied to objects and that's what frigate works off of. Recordings can be limited to motion but there's no motion events. In a future version there will be a timeline on the recordings page which shows when motion occurs.

Also side note if that's a motion mask everywhere but that one spot, object detection won't work well at all

**Reply by klutzzykitty** [reply]
Yeah I agree. 

I was trying it indoors before enabling it on my out-door camera. 

In my situation, I wanted to at-least have one automation upon motion detection when someone tosses garbage bags in the backyard. 
Person detection isn't reliable as the person is behind a wall. 
Was theorising if the motion caused by the bag falling over the wall can be picked up and an alert generated. 

Im able to pick up the motion caused by the tossed bag. Is there anything that can be triggered when this happens? 
Ex: An MQTT message being generated etc. 

In the discussions I've seen people talk about automating garden lights using Frigate upon motion detection. 
It would be working in a similar way isn't it ?

**Reply by NickM-27** [reply]
Yes there is a motion only sensor in HA / mqtt https://docs.frigate.video/integrations/mqtt#frigatecamera_namemotion

**Reply by klutzzykitty** [reply]
Right ! Lol
Completely forgot this existed. 

Thanks for reminding.

**Reply by will-riley** [reply]
"Motion is really not reliable as it's just a change in pixels, lights get turned on and the entire stream is one big glob of motion."

Respectfully disagree. That's an assumption about the exact circumstances of a camera deployment, the contents and action of the frame, and the intentions of the user. 
 
Upvoting motion as a "first class citizen", same as objects.

That said, Frigate is largely fantastic. Thanks!

**Reply by NickM-27** [reply]
You can disagree, but that's the way it is implemented in frigate. Frigate has a simpler form of motion detection and the goal is not to implement a more advanced (and more resource intensive) motion detection. Not sure if you've seen the new 0.14 UI but there is a tab for reviewing motion now.

**Reply by will-riley** [reply]
I see. I'm migrating from Motion-project (app), and it's got some very useful motion tuning options. Works well for it's scope, and the CPU drain is fully manageable with 6 HD cams and a decent CPU. Frigate seems to be putting all the eggs in the AI object-detection basket by contrast...Which is theoretically superior, but if there's no model for what you need to trigger on -- Pay up and spend the time isolating and prepping images for a new model. I understand now, thanks.

**Reply by will-riley** [reply, marked as answer]
Frigate 0.14 solves all things for me. Very well done indeed! Motion settings are totally adequate.

**Comment by alexsahka** [comment]
> Hello ! I was trying to figure out how to enable events to be generated upon motion detection happening within a zone.
> 
> Is there a way to generate an event based on motion being detected, irrespective of what object it is? Went through the docs explaining the [different modes and configuration](https://docs.frigate.video/configuration/record/#what-do-the-different-retain-modes-mean) but couldn't get it.
> 
> ` events: # Optional: Number of seconds before the event to include (default: shown below) retain: # Required: Default retention days (default: shown below) mode: motion`
> 
> I suppose the above part only effects what gets recorded and not the events generated.
> 
> Below is my zone and surrounding motion masks. ![cam_view](https://user-images.githubusercontent.com/118447577/208675120-465efae6-1831-438f-b234-72b141d0d277.png)
> 
> My goal is to generate an alert upon some motion being detected within the zone. As far as I've understood Frigate, the events and snapshots are only generated by detected objects isn't it ? And motion is just used as a primary check to send only that portion of the frame for object detection.
> 
> So, what would an example _config.yml_ look like if one wants to just detect motion within a zone? And then maybe generate a snapshot and video of it ?

Could you please help me figure out how to get motion events working in zones? I'm running Frigate 0.14. My understanding is that motion events in zones will also be reflected in MQTT?
